backend/schema/validate.go

```diff
@@ -281,6 +281,22 @@ func ValidateModule(module *Module) error {
 		return next()
 	})
 	merr = cleanErrors(merr)
+	sort.SliceStable(module.Decls, func(i, j int) bool {
+		priority := map[reflect.Type]int{
+			reflect.TypeOf(&Enum{}): 1,
+			reflect.TypeOf(&Data{}): 2,
+			reflect.TypeOf(&Verb{}): 3,
+		}
+		typeI := reflect.TypeOf(module.Decls[i])
+		typeJ := reflect.TypeOf(module.Decls[j])
+		if priority[typeI] == priority[typeJ] {
```

Comment:
Rather than the priority map, how about we just sort by the type name?

If you do want to keep the priority map, make it a global and build it automatically from `declUnion`.

Reply:
true, type name works - we can always update if priority and type name fall out of alignment